mailpoet/lib/Util/License/Features/Subscribers.php

```diff
@@ -18,22 +19,30 @@ class Subscribers {
   const PREMIUM_EMAIL_VOLUME_LIMIT_SETTING_KEY = 'premium.premium_key_state.data.email_volume_limit';
   const PREMIUM_EMAILS_SENT_SETTING_KEY = 'premium.premium_key_state.data.emails_sent';
   const PREMIUM_SUPPORT_SETTING_KEY = 'premium.premium_key_state.data.support_tier';
+  const SUBSCRIBERS_COUNT_CACHE_KEY = 'mailpoet_subscribers_count';
+  const SUBSCRIBERS_COUNT_CACHE_EXPIRATION_MINUTES = 60;
+  const SUBSCRIBERS_COUNT_CACHE_MIN_VALUE = 1000;
```

Comment:
In effect we cache an SQL query. Why don't we cache it always and start only above this threshold? To me it seems introducing an unnecessary complication.

Reply:
What do you mean by caching always? Moving it to a repository? Isn't it caching always now?

Reply:
I mean we cache the value only, when it is `> 1000` right? And I wonder why. https://github.com/mailpoet/mailpoet/pull/4312/files#diff-2d1730f301f3e90af93f2d5482c6446631bfc10e95b0743cba6ed3f0e4da4b2dR69-R71

Reply:
Ahh, right. The idea was to provide immediate feedback on count changes when the plugin is freshly installed and tested. Some people may try to check if everything works and some delayed info could be confusing. I also thought it was less of a BC break so safer not to break stuff — simply caching only for large numbers of subscribers. Do you think we should remove that logic? I'd feel a bit safer about these changes keeping it, I guess.